src/hooks/useRefObjectAsForwardedRef.ts

```diff
@@ -0,0 +1,12 @@
+import {ForwardedRef, RefObject, useImperativeHandle} from 'react'
+
+/**
+ * Use a ref object as the imperative handle for a forwarded ref. This can be used to
+ * synchronize the ref object with the forwarded ref and allow local access the reference
+ * instance with `.current`.
+ *
+ * **NOTE**: The `refObject` should be passed to the underlying element, NOT the `forwardedRef`.
+ */
+export function useRefObjectAsForwardedRef<T>(forwardedRef: ForwardedRef<T>, refObject: RefObject<T>): void {
```

Comment:
I'm open to suggestions for naming this. I think this name works but it could maybe be clearer. Other options I considered include:
- `useForwardedRefObject`
- `useForwardedRefAsRefObject`
- `useRefObjectAsHandle`